packages/theming/win32-theme/src/NativeModule/getThemingModule.native.ts

```diff
@@ -0,0 +1,31 @@
+import { OfficeThemingModule } from './officeThemingModule';
+import { NativeEventEmitter, NativeModules } from 'react-native';
+import { fallbackGetPalette, fallbackOfficeModule } from './fallbackOfficeModule';
+
+declare module 'react-native' {
+  interface NativeModulesStatic {
+    Theming: OfficeThemingModule & EventSubscriptionVendor;
+  }
+}
+
+/**
+ *  If we have a userAgent string, let's assume we're web debugging.  __DEV__ is for developer bundles.  Currently,
+ *  react-native only polyfills navigator with { product: 'ReactNative', geolocation: NativeModules.Geolocation }
+ */
+function disableGetPalette(): boolean {
+  const disabled = __DEV__ && navigator && navigator.userAgent !== undefined;
+  disabled && console.warn(console.warn('Web Debugging forces Theming Native Module to fallback to fake color values.'));
+  return disabled;
+}
+
+export function getThemingModule(): [OfficeThemingModule, NativeEventEmitter | undefined] {
+  const module = NativeModules && NativeModules.Theming;
+  // if the native module exists return the module + an emitter for it
+  if (module) {
+    // mock getPalette if it should be disabled, otherwise return the module directly
+    return [disableGetPalette() ? { ...module, getPalette: fallbackGetPalette } : module, new NativeEventEmitter(module)];
```

Comment:
So this does leave the NativeModule intact like the old version did, but you're still using a different fallbackGetPalette.

The old version did a require on the debugpalette.json you deleted.  That json was more fully specified than the impl in the fallbackThemingModule code, and it means that key-value pairs in the palette will disappear if we don't update the fallbackGetPalette to include the rest of the entries in the .json.

Now that I'm thinking about it, though, maybe we should get rid of the .json and export it from the NM as a constant.  I think I probably did it this way to avoid a second check in to the native code at the time.

Reply:
I think both the json and the code in the fallback have all the keys specified, there's an open question of which one is more correct, but they both should work.

Reply:
I did a diff of the files and added the types to the OfficePalette type, then patched the keys from the debugjson.json file. We probably do want to copy a full standard White or Colorful theme definition to ensure that when the fallbacks happen they are doing the right thing.